24/25/28 file structure and database

RamyAlshurafa opened the pull request:
#### Issues this PR relates to
#24 
#25 
#28

numbered skills (will make the review easier when reviewing the skills and skill areas initial data (in skill-areas-skills table)

[Volunteer digital skills framework (1).xlsx](https://github.com/yalla-coop/scouts/files/5836625/Volunteer.digital.skills.framework.1.xlsx)
---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install
- `npm run build:db`
- then using psql/pgcli/pgAdmin check the data been created
---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars

---

#### Quick Pull Request Checklist
Please make sure you ensure the following has been done before marking your PR as awaiting-review. For full checklist and process review, please refer to Process Guide (#8)
- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- The new work has been checked with Axe accessibility checker and doesn't throw significant errors  
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs

thejoefriel commented on server/database/models/activities/schema.sql:
```diff
@@ -0,0 +1,19 @@
+  "title" VARCHAR NOT NULL,
+  "difficulty" INTEGER NOT NULL,
+  "completion_time" INTEGER NOT NULL,
+  "description" TEXT NOT NULL,
```
In the database after the build, I'm not getting any descriptions stored. I'm wondering if it's because this is text and it should be varchar? Not sure but I can see you've put description in the dummy data so there seems to be an issue here

RamyAlshurafa replied:
can you double-check again please

thejoefriel replied:
Hmm, it's not showing in PSequel but I ran the command in my terminal and it is there, so I'm assuming it's a PSequel quirk. Happy to mark this as resolved

thejoefriel commented on server/database/dummy-data/skills.js:
```diff
@@ -0,0 +1,464 @@
+
+    {
+      title: 'I can turn on a device',
+      code: 'TURN_ON_DEVICE',
```
I noticed in the database that we've got an icon data field but we haven't stored an icon names - is that right?

RamyAlshurafa replied:
yeah we should have, i added a "default" icon for all of them

thejoefriel commented on server/database/dummy-data/users.js:
```diff
@@ -1,16 +1,111 @@
+      password,
+      organisation: null,
+      selectedPath: learningPaths.BASIC,
+      assessmentScore: 2050,
```
The score should really be up to max 100 so for these dummy ones, can we put them at under 100 to make sure we get an accurate looking score to track the progress. 
Same for improvement score

RamyAlshurafa replied:
done

thejoefriel commented on server/database/models/users-check-ins/schema.sql:
```diff
@@ -0,0 +1,14 @@
+DROP TABLE IF EXISTS "users_checks_ins" CASCADE;
```
please call this 'user_check_ins' instead

thejoefriel replied:
also should we have some dummy data for this?

RamyAlshurafa replied:
done

thejoefriel commented on server/database/dummy-data/users-completed-activities.js:
```diff
@@ -0,0 +1,46 @@
+    {
+      userId: JohnReese.id,
+      activityId: microsoftWordActivity.id,
+      addedScore: 700,
```
can we make this a more realistic score (e.g. 3)

RamyAlshurafa replied:
this equal 7

thejoefriel commented on server/database/dummy-data/users-skills.js:
```diff
@@ -0,0 +1,66 @@
+      userId: JohnReese.id,
+      skillId: skills[2].id,
+      status: skillStatuses.COMPLETED,
+      addedScore: 1250,
```
same here as with the other scores, let's make more like 5

RamyAlshurafa replied:
this was meant to equal 12.5% because dealing with float numbers require us to format them into 2 decimals, but will change it to Real type

thejoefriel commented on server/constants/data-types.js:
```diff
@@ -0,0 +1,16 @@
+export const skillStatuses = {
+  ALREADY_HAS: 'ALREADY_HAS',
+  COMPLETED: 'COMPLETED',
+  QUEUED: 'QUEUED',
```
I was thinking about this a bit more, do we actually need queued? We definitely need already has and completed, but wouldn't we know all remaining skills from the relevant skill areas for the user that are not marked as either already has or completed?

RamyAlshurafa replied:
yeah you are right, fixed

thejoefriel commented on server/database/dummy-data/organisations.js:
```diff
@@ -0,0 +1,27 @@
+};
+
+const createOrganisations = async () => {
+  const AvonTyrrellOrg = await createOrganisation(['Avon Tyrrell']);
```
Let's make at least one of these Scouts since this will most likely be the only organisation using the platform in the medium term

RamyAlshurafa replied:
done

thejoefriel commented on server/constants/index.js:
```diff
@@ -1 +1,2 @@
 export * from './common';
+export * from './data-types';
```
I noticed skill codes or skill area codes aren't stored as constants. Should we do this so that there's one place where these can be checked/updated?

RamyAlshurafa replied:
good idea, done

thejoefriel reviewed (changes requested):
- activities seemed to get created twice. Is that right?
<img width="1920" alt="Screenshot 2021-01-19 at 17 06 01" src="https://user-images.githubusercontent.com/19708921/105068275-aa059500-5a78-11eb-8a8a-ba1ef11894e8.png">

- I don't think this was on your PR but the migrations.sql file I've noticed has a bunch of migrations from the Tempo project! Do you mind removing please?

RamyAlshurafa commented:
> activities seemed to get created twice. Is that right?
 
fixed

> I don't think this was on your PR but the migrations.sql file I've noticed has a bunch of migrations from the Tempo project! Do you mind removing please?

oops, fixed
